crates/iota-json-rpc-types/src/iota_transaction.rs

```
@@ -1264,6 +1265,80 @@ pub struct IotaExecutionResult {
+    /// Execution error from executing the move view call
+    #[serde(rename = "executionError")]
+    Error(String),
+    /// The return values of the move view function
+    #[serde(rename = "functionReturnValues")]
+    Results(Vec<IotaMoveValue>),
+}
+
+impl IotaMoveViewCallResults {
+    /// Process the dev-inspect results to produce the response
+    /// of the move-view function call.
+    pub async fn from_dev_inspect_results<S: PackageStore>(
+        package_store: S,
+        dev_inspect_results: DevInspectResults,
+    ) -> anyhow::Result<Self> {
+        if let Some(error) = dev_inspect_results.error {
+            return Ok(Self::Error(error));
+        }
+        let Some(mut tx_execution_results) = dev_inspect_results.results else {
+            return Ok(Self::Error("function call returned no values".into()));
+        };
+        let Some(execution_results) = tx_execution_results.pop() else {
+            return Ok(Self::Error(
+                "no results from move view function call".into(),
+            ));
+        };
+        if !tx_execution_results.is_empty() {
+            return Ok(Self::Error("multiple transactions executed".into()));
+        }
+        let mut move_call_results = Vec::with_capacity(execution_results.return_values.len());
+        let package_resolver = Resolver::new(package_store);
+        let mut execution_results =
+            execution_results.into_stream_return_value_layouts(&package_resolver);
+        while let Some(result) = execution_results.next().await {
+            let (bytes, move_type_layout) = result?;
+            let move_value = BoundedVisitor::deserialize_value(&bytes, &move_type_layout)?;
+            move_call_results.push(IotaMoveValue::from(move_value));
+        }
+        Ok(Self::Results(move_call_results))
+    }
+
+    pub fn into_return_values(self) -> Vec<IotaMoveValue> {
+        match self {
+            IotaMoveViewCallResults::Error(_) => Default::default(),
```

Isn't this hiding the error in case it happens? Could we return Result<Vec<_>> or Option<Vec<_>> from this fn?

---

An empty vector implies that there are no return values. Empty return values imply that it is an error, as a move view function must return values if successfully executed.